pkg/server/api_v2_databases_metadata_test.go

```diff
@@ -512,29 +519,161 @@ func TestGetTableMetadataUpdateJobStatus(t *testing.T) {
 		userClient, _, err := ts.GetAuthenticatedHTTPClientAndCookie(sessionUsername, false, 1)
 		require.NoError(t, err)
 
-		failed := makeApiRequest[interface{}](t, userClient, ts.AdminURL().WithPath(uri).String())
+		failed := makeApiRequest[interface{}](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, http.StatusText(http.StatusNotFound), failed)
 
 		_, e := conn.Exec(fmt.Sprintf("GRANT CONNECT ON DATABASE defaultdb TO %s", sessionUsername.Normalized()))
 		require.NoError(t, e)
 
-		mdResp := makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String())
+		mdResp := makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, "NOT_RUNNING", mdResp.CurrentStatus)
 
 		_, e = conn.Exec(fmt.Sprintf("REVOKE CONNECT ON DATABASE defaultdb FROM %s", sessionUsername.Normalized()))
 		require.NoError(t, e)
-		failed = makeApiRequest[string](t, userClient, ts.AdminURL().WithPath(uri).String())
+		failed = makeApiRequest[string](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, http.StatusText(http.StatusNotFound), failed)
 
 		_, e = conn.Exec(fmt.Sprintf("GRANT admin TO %s", sessionUsername.Normalized()))
 		require.NoError(t, e)
-		mdResp = makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String())
+		mdResp = makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, "NOT_RUNNING", mdResp.CurrentStatus)
 	})
 }
 
-func makeApiRequest[T any](t *testing.T, client http.Client, uri string) (mdResp T) {
-	req, err := http.NewRequest("GET", uri, nil)
+func TestTriggerMetadataUpdateJob(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	skip.UnderStress(t, "too slow under stress")
+
+	testCluster := serverutils.StartCluster(t, 3, base.TestClusterArgs{})
+	ctx := context.Background()
+	defer testCluster.Stopper().Stop(ctx)
+	conn := testCluster.ServerConn(0)
+	defer conn.Close()
+	ts := testCluster.Server(0)
+
+	// Get the node id that claimed the update job. We'll issue the
+	// RPC to a node that doesn't own the job to test that the RPC can
+	// propagate the request to the correct node.
+	var nodeID int
+	testutils.SucceedsSoon(t, func() error {
+		row, err := conn.Query(`
+SELECT claim_instance_id FROM system.jobs 
+WHERE id = $1 AND claim_instance_id IS NOT NULL`, jobs.UpdateTableMetadataCacheJobID)
+		require.NoError(t, err)
+		if !row.Next() {
+			return errors.New("no node has claimed the job")
+		}
+		require.NoError(t, row.Scan(&nodeID))
+
+		rpcGatewayNode := (nodeID + 1) % 3
+		_, err = testCluster.Server(rpcGatewayNode).GetStatusClient(t).UpdateTableMetadataCache(ctx,
+			&serverpb.UpdateTableMetadataCacheRequest{Local: false})
```

Comment:
I think we need to delete this part which is triggering via status server.

Reply:
Removed this and switched to using a test knob to indicate when the job is ready